File: bt2.9trang34.cpp
#include <bits/stdc++.h>
using namespace std;
int gcd(int a,int b){
	if (b==0) return a;
	return gcd(b,a%b);
}
int lcm(int a,int b){
	int x=a*b;
	return x/gcd(a,b);
}
int main(){
	int n;cin>>n;vector<int> a(n);
	for (int i=0;i<n;i++) cin>>a[i];
	int x=a[0];
	for (int i=1;i<n;i++){
		x=lcm(x,a[i]);
	}
	cout<<x;
}

File: bt2.9trang34.py
def gcd(a,b):
    if (b==0):
        return a
    return gcd(b,a%b)
def lcm(a,b):
    x=a*b
    return x//gcd(a,b)
a = list(map(int,input().split()))
x=a[0]
y=a[0]
for i in range(1,len(a)):
    y=gcd(y,a[i])
print(y)
for i in range(1,len(a)):
    x=lcm(x,a[i])
print(x)